src/wowman/nfo.clj

```diff
@@ -44,33 +55,61 @@
 
 (defn-spec write-nfo ::sp/extant-file
   "given an installation directory and an addon, extract the neccessary bits and write them to a nfo file"
-  [install-dir ::sp/extant-dir, addon ::sp/addon-or-toc-addon, addon-dirname string?, primary? boolean?]
+  [install-dir ::sp/extant-dir, addon ::sp/nfo-input-minimum, addon-dirname string?, primary? boolean?, game-track ::sp/game-track]
   (let [path (nfo-path install-dir addon-dirname)]
-    (utils/dump-json-file path (derive addon primary?))
+    (utils/dump-json-file path (derive addon primary? game-track))
     path))
 
+(defn-spec update-nfo ::sp/extant-file
+  "refreshes the nfo data for the given addon."
+  [install-dir ::sp/extant-dir, addon (s/merge ::sp/nfo-input-minimum (s/keys :opt [::sp/version]))]
+  (let [;; important! as an addon is updated or installed, the `:installed-version` is overridden by the `:version`
+        ;; we don't want to alter the version it thinks is installed here
+        addon (merge addon {:version (:installed-version addon)})]
+    (write-nfo install-dir addon (:dirname addon) (:primary? addon) (:game-track addon))))
+
 (defn-spec rm-nfo nil?
   "deletes a nfo file and only a nfo file"
   [path ::sp/extant-file]
   (when (= nfo-filename (fs/base-name path))
     (fs/delete path)
     nil))
 
-(defn-spec read-nfo-file (s/or :ok ::sp/nfo, :error nil?)
-  "reads the nfo file. 
-  failure to load the json results in the file being deleted. 
+(defn-spec has-nfo? boolean?
+  "returns true if certain keys are detected in addon data."
+  [install-dir ::sp/extant-dir, addon any?]
```

Comment:
any? here is lazy. we want a map with a `dirname`, that's all

Reply:
ehhhh ... what I have now confuses me. I need to revisit *all* of the typing I've done. `any?` in a spec is a smell anyway, I'll know to revisit it once things are sorted